### PR TITLE
[KOGITO-2264] - Upgrade Operator SDK to 0.18.0

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,1352 @@
...
+              {
+                "name": "MP_MESSAGING_OUTGOING_KOGITO_PROCESSINSTANCES_EVENTS_BOOTSTRAP_SERVERS"
+              },
+              {
+                "name": "MP_MESSAGING_OUTGOING_KOGITO_USERTASKINSTANCES_EVENTS_BOOTSTRAP_SERVERS"
+              }
+            ]
+          }
+        }
+      ]
```
Comment (ricardozanini) [inline deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml:498]: Please remove the `deploy/examples` directory, move it to the parent directory. On this new version, the SDK is considering every yaml file on `deploy` to be an example CR.

File: go.mod [hunk 2]
```diff
@@ -21,35 +22,34 @@ require (
...
+	k8s.io/apiextensions-apiserver v0.17.4
+	k8s.io/apimachinery v0.18.4
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/kube-openapi v0.0.0-20190918143330-0270cf2f1c1d
-	sigs.k8s.io/controller-runtime v0.4.0
+	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a
+	sigs.k8s.io/controller-runtime v0.5.2
 )
 
 // Pinned to kubernetes-1.16.2
```
Comment (ricardozanini) [inline go.mod:45]: Isn't this comment wrong? I believe we are now on 17?

File: go.mod [hunk 3]
```diff
@@ -21,35 +22,33 @@ require (
 	github.com/matryer/is v1.2.0 // indirect
 	github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
 	github.com/openshift/client-go v0.0.0-20190923180330-3b6373338c9b
-	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20191115003340-16619cd27fa5
+	github.com/operator-framework/operator-lifecycle-manager v0.0.0-20200321030439-57b580e57e88
 	github.com/operator-framework/operator-marketplace v0.0.0-20190919183128-4ef67b2f50e9
-	github.com/operator-framework/operator-sdk v0.15.1
-	github.com/prometheus/common v0.7.0
+	github.com/operator-framework/operator-sdk v0.17.0
```
Comment (ricardozanini) [inline go.mod:27]: The latest version is `0.18.2`. :)

**Comment by spolti** [comment]
@sutaakar @Kaitou786 do you guys mind to test the operator as well? quick local tests and bdd tests didn't report any issues so far :)

File: pkg/controller/kogitobuild/build/trigger.go [hunk 4]
```diff
@@ -47,7 +48,7 @@ func StartNewBuild(buildConfig *v1.BuildConfig, client *client.Client) error {
 
 // cancelRunningBuilds cancels any running builds for the given BuildConfig
 func cancelRunningBuilds(buildConfig *v1.BuildConfig, client *client.Client) error {
-	builds, err := client.BuildCli.Builds(buildConfig.Namespace).List(
+	builds, err := client.BuildCli.Builds(buildConfig.Namespace).List(context.TODO(),
```
Comment (ricardozanini) [inline pkg/controller/kogitobuild/build/trigger.go:51]: This parameter can be inside the `openshift/build.go` instead. We will always use `context.TODO()`. That's the same behavior of others client wrappers we have as well. Once we change this behavior to allow a change in the context input parameter, we might revaluate. :P

File: pkg/controller/kogitobuild/build/trigger.go [hunk 5]
```diff
@@ -64,12 +65,12 @@ func cancelRunningBuilds(buildConfig *v1.BuildConfig, client *client.Client) err
 				defer wg.Done()
 				err := wait.Poll(poolWaitTimeout, cancelUpdateTimeout, func() (bool, error) {
 					build.Status.Cancelled = true
-					_, err := client.BuildCli.Builds(build.Namespace).Update(build)
+					_, err := client.BuildCli.Builds(build.Namespace).Update(context.TODO(), build, metav1.UpdateOptions{})
```
Comment (ricardozanini) [inline pkg/controller/kogitobuild/build/trigger.go:68]: Same here, `UpdateOptions` is unnecessary in our context for now.

**Comment by ricardozanini** [comment]
> One last thing regarding wrapping the optional new parameters introduced by the openshit client library.

Ignore. I should have used the `openshift` wrapper instead of calling the `BuildCli` directly in that PR. I'll fix next time we have a change on `KogitoBuild`. :)

**Comment by Kaitou786** [comment]
Ran the smoke and full tests with the upgraded operator-sdk image on AWS Jenkins, the smoke tests succeeded and full tests have some failure(which should be the case as less resources)
`Smoke test logs: `  https://jenkins-kogito-tools.apps.kogito.automation.rhmw.io/blue/organizations/jenkins/Kogito%2Fkogito-cloud-operator-pr/detail/PR-399/17/pipeline

`Full test logs: `  https://jenkins-kogito-tools.apps.kogito.automation.rhmw.io/blue/organizations/jenkins/Kogito%2Fkogito-cloud-operator-pr/detail/PR-399/18/pipeline

PS: In full tests, the stage name is `smoke test` but they are running the full tests, I forgot to change the stage name, but you can see the command.

**Comment by sutaakar** [comment]
From my local experiments it seems to work as expected.

**Pull request merged**